src/main/rx/expresslrs_common.c

```diff
@@ -0,0 +1,642 @@
+        }
+    }
+
+    for (uint8_t i = 0; i < seqCount; i++) {
+        // if it's not the sync channel
+        if (i % numFreqs != 0) {
+            uint8_t offset = (i / numFreqs) * numFreqs; // offset to start of current block
+            uint8_t rand = rngN(numFreqs - 1) + 1; // random number between 1 and numFreqs
+
+            // switch this entry and another random entry in the same block
+            uint8_t temp = FHSSsequence[i];
+            FHSSsequence[i] = FHSSsequence[offset + rand];
+            FHSSsequence[offset + rand] = temp;
+        }
+    }
+}
+
+uint8_t tlmRatioEnumToValue(const elrs_tlm_ratio_e enumval)
+{
+    switch (enumval)
+    {
+    case TLM_RATIO_NO_TLM:
+        return 1;
+        break;
+    case TLM_RATIO_1_2:
+        return 2;
+        break;
+    case TLM_RATIO_1_4:
+        return 4;
+        break;
+    case TLM_RATIO_1_8:
+        return 8;
+        break;
+    case TLM_RATIO_1_16:
+        return 16;
+        break;
+    case TLM_RATIO_1_32:
+        return 32;
+        break;
+    case TLM_RATIO_1_64:
+        return 64;
+        break;
+    case TLM_RATIO_1_128:
+        return 128;
+        break;
+    default:
+        return 0;
+    }
+}
+
+uint16_t rateEnumToHz(const elrs_rf_rate_e eRate)
+{
+    switch (eRate)
+    {
+    case RATE_500HZ: return 500;
+    case RATE_250HZ: return 250;
+    case RATE_200HZ: return 200;
+    case RATE_150HZ: return 150;
+    case RATE_100HZ: return 100;
+    case RATE_50HZ: return 50;
+    case RATE_25HZ: return 25;
+    case RATE_4HZ: return 4;
+    default: return 1;
+    }
+}
+
+uint16_t txPowerIndexToValue(const uint8_t index)
+{
+    switch (index)
+    {
+    case 0: return 0;
+    case 1: return 10;
+    case 2: return 25;
+    case 3: return 100;
+    case 4: return 500;
+    case 5: return 1000;
+    case 6: return 2000;
+    case 7: return 250;
+    case 8: return 50;
+    default: return 0;
+    }
+}
+
+#define ELRS_LQ_DEPTH 4 //100 % 32
+
+typedef struct linkQuality_s {
+    uint32_t array[ELRS_LQ_DEPTH];
+    uint8_t value;
+    uint8_t index;
+    uint32_t mask;
+} linkQuality_t;
+
+static linkQuality_t lq;
+
+void lqIncrease(void)
+{
+    if (lqPeriodIsSet()) {
+        return;
+    }
+    lq.array[lq.index] |= lq.mask;
+    lq.value += 1;
+}
+
+void lqNewPeriod(void)
+{
+    lq.mask <<= 1;
+    if (lq.mask == 0) {
+        lq.mask = (1 << 0);
+        lq.index += 1;
+    }
+
+    // At idx N / 32 and bit N % 32, wrap back to idx=0, bit=0
+    if ((lq.index == 3) && (lq.mask & (1 << ELRS_LQ_DEPTH))) {
+        lq.index = 0;
+        lq.mask = (1 << 0);
+    }
+
+    if ((lq.array[lq.index] & lq.mask) != 0) {
+        lq.array[lq.index] &= ~lq.mask;
+        lq.value -= 1;
+    }
+}
+
+uint8_t lqGet(void)
+{
+    return lq.value;
+}
+
+bool lqPeriodIsSet(void)
+{
+    return lq.array[lq.index] & lq.mask;
+}
+
+void lqReset(void)
+{
+    memset(&lq, 0, sizeof(lq));
+    lq.mask = (1 << 0);
+}
+
+inline uint16_t convertSwitch1b(const uint16_t val)
```

Comment:
Probably best to let the compiler decide on optimisation. Any evidence that this is needed?

Reply:
removed inline